src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -436,7 +436,8 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     //   region to change, such as when new text enters the buffer or the viewport is scrolled
     void ControlCore::UpdatePatternLocations()
     {
-        _terminal->UpdatePatterns();
+        auto lock = _terminal->LockForWriting();
+        _terminal->UpdatePatternsUnderLock();

```

Comment:
Just curious, why'd you split this up?

Reply:
Calling `UpdatePatterns` from `UpdateSettings` was causing a hang because of the writing lock, and I didn't want to have _both_ an `UpdatePatterns` and an `UpdatePatternsUnderLock`. And since this is the only other place we ask Terminal to update the pattern locations, I thought it'd be fine to ask it to grab the lock